common/src/main/java/com/hedera/block/common/utils/Translator.java

```diff
@@ -0,0 +1,218 @@
+            @NonNull final SubscribeStreamResponse subscribeStreamResponse) {
+        try {
+            final byte[] pbjBytes =
+                    asBytes(SubscribeStreamResponse.PROTOBUF, subscribeStreamResponse);
+            return com.hedera.hapi.block.protoc.SubscribeStreamResponse.parseFrom(pbjBytes);
+        } catch (final InvalidProtocolBufferException e) {
+            final String message =
+                    INVALID_BUFFER_MESSAGE.formatted(
+                            "SubscribeStreamResponse", subscribeStreamResponse);
+            LOGGER.log(ERROR, message);
+            throw new RuntimeException(message, e);
+        }
+    }
+
+    /**
+     * Converts a {@link SubscribeStreamRequest} to a {@link
+     * com.hedera.hapi.block.protoc.SubscribeStreamRequest}.
+     *
+     * @param subscribeStreamRequest the {@link SubscribeStreamRequest} to convert
+     * @return the converted {@link com.hedera.hapi.block.protoc.SubscribeStreamRequest}
+     */
+    @NonNull
+    public static com.hedera.hapi.block.protoc.SubscribeStreamRequest fromPbj(
+            @NonNull final SubscribeStreamRequest subscribeStreamRequest) {
+        try {
+            final byte[] pbjBytes =
+                    asBytes(SubscribeStreamRequest.PROTOBUF, subscribeStreamRequest);
+            return com.hedera.hapi.block.protoc.SubscribeStreamRequest.parseFrom(pbjBytes);
+        } catch (final InvalidProtocolBufferException e) {
+            final String message =
+                    INVALID_BUFFER_MESSAGE.formatted(
+                            "SubscribeStreamRequest", subscribeStreamRequest);
+            LOGGER.log(ERROR, message);
+            throw new RuntimeException(message, e);
+        }
+    }
+
+    /**
+     * Converts protoc bytes to a PBJ record of the same type.
+     *
+     * @param <T> the type of PBJ record to convert to
+     * @param codec the record codec to convert the bytes to a PBJ record
+     * @param bytes the protoc bytes to convert to a PBJ record
+     * @return the converted PBJ record
+     * @throws ParseException if the conversion between the protoc bytes and PBJ objects fails
+     */
+    @NonNull
+    public static <T extends Record> T toPbj(
+            @NonNull final Codec<T> codec, @NonNull final byte[] bytes) throws ParseException {
+        requireNonNull(codec);
+        requireNonNull(bytes);
+        return codec.parse(Bytes.wrap(bytes));
+    }
+
+    @NonNull
+    private static <T extends Record> byte[] asBytes(
+            @NonNull final Codec<T> codec, @NonNull final T tx) {
+        requireNonNull(codec);
+        requireNonNull(tx);
+        try (final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                final WritableStreamingData wsd = new WritableStreamingData(baos)) {
+            // TODO: should 'baos' and 'wsd' be used like that (autoclosed) here? Probably yes, but
+            // maybe something I do not know?
```

Comment:
TODO, let's discuss.

Reply:
@AlfredoG87 do you have an opinion on this?

Reply:
I think they should be indeed be closed.

Reply:
this file will be deleted

Reply:
this file will be deleted